### PR TITLE
add end date constraint 

### LINKED ISSUE #19: Add functionality to terminate the staking program at a certain date as rewards are not going to be distributed forever

### LINKED ISSUE #23: Set end date to 1768993200 in deployment

### PR DESCRIPTION
**Pull request by perionjames**
Closes #19 
Closes #23

### CODE REVIEW

**Review by amosperion — APPROVED** [review]
Looks good to me, @geoshak want to take a look too

File: scripts/Deployment.ts [hunk 1]
```diff
@@ -37,7 +37,10 @@ const ESCROW_PORTION = parseEther("1");
 const MAX_BONUS = parseEther("5");
 
 // Maximum duration that a lock can have
-const MAX_LOCK_DURATION = 60 * 60 * 24 * 365 * 4;
+const MAX_LOCK_DURATION = 60 * 60 * 24 * 365 * 3;
```
Comment (perionjames) [inline scripts/Deployment.ts:40]: Set to 3 years here and we use this figure in the curve calculation. Note that if the end date ends up being earlier than 3 years from the time of launch, nobody will be able to get the full multiplier.

File: test/Upgradeable.ts [hunk 2]
```diff
@@ -389,11 +392,11 @@ describe("TimeLockPool", function () {
...
-    
-   
 
 
-    
-    
+
+
+
+
```
Comment (geoshak) [inline test/Upgradeable.ts:400]: Are all these blank lines here on purpose?

File: scripts/Deployment.ts (hunk 1 above)
Comment (geoshak) [reply scripts/Deployment.ts:40]: I saw that the end date is 1/21/2026, 8:00:00 AM
Should we allow some extra room then?
So people can get the full multiplier if they want, on the first few weeks at least?

**Review by geoshak — APPROVED** [review]

File: test/Upgradeable.ts (hunk 2 above)
Comment (perionjames) [reply test/Upgradeable.ts:400]: Looks like it is the prettier.

File: scripts/Deployment.ts (hunk 1 above)
Comment (perionjames) [reply scripts/Deployment.ts:40]: We will likely switch back to a linear model.

**Pull request merged**